### ISSUE TITLE
Spaces around equal

### ISSUE THREAD

**Issue by leusrox** [yannickcr/eslint-plugin-react#2334]
Expected a warning for this example, when initialize the state as class field declaration:
```
export default class Some extends Component {
  state={}
  ...

  componentDidMount() {
    const test={}
  }
```

`space-infix-ops` and `semi` show warnings for `test` variable, but ignore `state`.

Is it possible to show warnings also for `state`?

**Comment by golopot** [comment]
This is probably a wrong repo for the issue. `jsx-equals-spacing` only concern about spaces in jsx. The example is not currently covered by eslint core because that syntax, class field, is not yet standard ecmascript (it is a stage 3 proposal). And as per https://github.com/eslint/eslint#what-about-experimental-features, core rules will not support experimental syntax.

**Comment by ljharb** [comment]
This is exactly right; when it’s stage 4, eslint itself will have to change to support this.
